corehq/blobs/util.py

```diff
@@ -37,6 +39,57 @@ def pre_init(self, value, obj):
         return self.get_default() if value is None else value
 
 
+# adapted from https://stackoverflow.com/a/31566082
+class Buffer(object):
+    def __init__(self):
+        self.__buf = deque()
+        self.__size = 0
```

Comment:
Private members should be prefixed with one underscore, not two. Name mangling (invoked by dunder-prefix) is nasty and should be avoided if possible. The same applies to `GzipCompressReadStream` below.